app/src/main/java/org/kinecosystem/kinit/server/TaskService.kt

```diff
@@ -119,19 +137,27 @@ class TaskService(context: Context, api: TasksApi,
 
                         if (response != null && response.isSuccessful) {
                             Log.d("TaskService", "submit onResponse: ${response.body()}")
-                            tasksRepo.taskState = TaskState.SUBMITTED_SUCCESS_WAIT_FOR_REWARD
+                            categoriesRepository.currentTaskRepo?.taskState = TaskState.SUBMITTED_SUCCESS_WAIT_FOR_REWARD
```

Comment:
you can not be 100% sure that currentTaskRepo has not been changed while waiting for the submit response.

Reply:
done